### PR TITLE
allow use of x-* service names as depends_on keys

### PR DESCRIPTION
**Pull request by ndeloof**
fix https://github.com/docker/compose/issues/12171

### CODE REVIEW

File: loader/loader_test.go [hunk 1]
```diff
@@ -3409,3 +3409,24 @@ services:
...
+
+func TestLoadDependsOnX(t *testing.T) {
+	p, err := loadYAML(`
+name: load-depends-on-x
+services:
+  test:
+    image: test
+    depends_on:
+      - x-foo
+  foo:
```
Comment (glours) [inline loader/loader_test.go:3421]: It should be `x-foo`, no?
```suggestion
  x-foo:
```

Comment (ndeloof) [reply loader/loader_test.go:3421]: indeed, not actually used by the test

**Review by glours — APPROVED** [review]
LGTM

**Review by jhrotko — APPROVED** [review]
LGTM

**Pull request merged**